### ISSUE TITLE
[BUG] Containers should run as non-root

### ISSUE THREAD

**Issue by uhthomas** [immich-app/immich#2750]
### The bug

Containers currently run as root, which is bad container security practise and causes problems like mentioned [here](https://github.com/immich-app/immich/issues/2740#issuecomment-1588157854).

```sh
❯ docker run -it ghcr.io/immich-app/immich-server:v1.60.0
/usr/src/app # id
uid=0(root) gid=0(root) groups=0(root),1(bin),2(daemon),3(sys),4(adm),6(disk),10(wheel),11(floppy),20(dialout),26(tape),27(video)
```

Immich is perfectly capable of running as non-root, as I do so with security contexts in my [Kubernetes cluster](https://github.com/uhthomas/automata/blob/6ec58b723bb6a34f005066e4546f25f344a54e09/k8s/unwind/immich/immich_server/deployment_list.cue#L115-L128). I believe the nginx container is already unprivileged, so it should just be a case of following the official [non-root guide](https://github.com/nodejs/docker-node/blob/1b2e8eb26c393986426ad5d88a1e190b93534906/docs/BestPractices.md#non-root-user) for node images.

### The OS that Immich Server is running on

N/A

### Version of Immich Server

v1.60.0

### Version of Immich Mobile App

N/A

### Platform with the issue

- [X] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
N/A
```

### Your .env content

```Shell
N/A
```

### Reproduction steps

```bash
N/A
```

**Comment by klaus1k** [comment]
Until this is implemented, is there a recommended way to run immich as non-root with docker compose?

I just added the following to every container: 
```
environment:
  - PUID=1000
  - GUID=1000 
```

Is there a more elegant way?

Thanks in advance!

**Comment by BenRoe** [comment]
> I changed user and group by simply declaring the `user` property in the docker compose file and it seemed to work.

Here a example for docker newbies 

```yaml
...
services:
  immich-server:
    container_name: immich_server
    image: ghcr.io/immich-app/immich-server:${IMMICH_VERSION:-release}
    user: "1000:1000"
...
```

**Comment by kamzil** [comment]
Even if you set user to non-root with `user` option in docker-compose.yml, all the files in the volumes are still owned by root user. See #4903

**Comment by BenRoe** [comment]
> Even if you set user to non-root with `user` option in docker-compose.yml, all the files in the volumes are still owned by root user, e.g. the cache files in the volume created by immich_redis container. See #4903

If you set "mount a volume" in the docker-compose and this folder does not exist, docker will create it as a root. That's normal docker behaviour.

Solution:

1. Create the volume folder before the first start the docker compose with the correct user
2. If you already startet the docker compose, stop it and change the folder owner with `sudo chown -R 1000:1000 /data`

**Comment by bphenriques** [comment]
I was running with `user` and then migrated to a rootless `docker` installation ([docs](https://docs.docker.com/engine/security/rootless/)). It was working before with `user` but not anymore with rootless `docker` instalation. Exceptions related with permissions in the microservice:
```
ERROR [ExceptionHandler] could not open file "global/pg_filenode.map": Permission denied
```

I checked the directories and the user owns the directories so I am not exactly what I did wrong.

**Comment by kamzil** [comment]
Using non-anonymous and explicit volumes instead of anonymous together with pre-creating or chowning a folder worked for me. Thanks @BenRoe for the tip.

**Comment by mmomjian** [comment]
I think this issue can be closed. We support running as non root: https://immich.app/docs/FAQ/#how-can-i-run-immich-as-a-non-root-user

changing the default would probably result in external libraries not working for a large number of users, and a very high non-Immich support burden. We offer the option for users who know how to configure permissions.

**Comment by Eirikr70** [comment]
Hello, I have tried to migrate from a rootfull setup to a rootless one this morning.I have copied all the existing volumes and changed the owner, and restarted in rootless mode. The problem I have encountered has been that Immich has considered that my setup was a new one and has started all anew, with the declaration of the administrator, etc. So all the previous pictures would have needed to be reimported. Is there a way to restart in rootless mode without actually restarting from scratch ?
